client/ayon_core/tools/sceneinventory/view.py

```diff
@@ -226,10 +230,12 @@ def _build_item_menu_for_selection(self, menu, indexes, active_index):
             # Keep remove action for invalid items
             menu.addAction(remove_action)
             return
-
-        version_items_by_product_id = self._controller.get_version_items(
-            product_ids
-        )
+        version_items_by_product_id = {}
+        for project_name, product_ids in project_products.items():
+            version_items_by_product_id.update(
+                self._controller.get_version_items(
+                project_name, product_ids)
+            )
```

Comment:
This resulting dict is technically invalid - a project id may be re-used across projects - it's not enforced to be unique.

Reply:
You meant product id? If yes, I would not bother (at least not now).